namerd/storage/consul/src/main/scala/io/buoyant/namerd/storage/consul/ConsulDtabStore.scala

```diff
@@ -19,6 +21,8 @@ class ConsulDtabStore(
 
   private[this] val log = Logger.get("consul")
 
+  def namespaceIsValid(ns: Ns): Boolean = Pattern.matches("[\\w+\\-?/?]+", ns)
```

Comment:
We should definitely build the regex statically so that it doesn't need to be compiled on every invocation of this method.  Regex invocation is slow.

```scala
val validNs = "[\\w+\\-?/?]+".r

def namespaceIsValid(ns: Ns): Boolean = validNs.matches(ns)
```

Or something like that

Reply:
Also, if we want a valid name to consist only of canonical DNS label characters, we already have a regex for matching them: 
https://github.com/linkerd/linkerd/blob/642e14fd7e7f0d961a023cbf55ecaa23b070f649/namer/core/src/main/scala/io/buoyant/hostport.scala#L12-L20
We could put this regex someplace where it's publicly visible and just reuse it...